os/arch/arm/src/imxrt/imxrt_clock.c

```diff
@@ -0,0 +1,1129 @@
+        /* Pll3_sw_clk ---> Periph_clk2_clk ---> Periph_clk */
+        case CCM_CBCMR_PERIPH_CLK2_SEL(0U):
+            freq = imxrt_clock_getpllfreq(kCLOCK_PllUsb1);
+            break;
+        /* Osc_clk ---> Periph_clk2_clk ---> Periph_clk */
+        case CCM_CBCMR_PERIPH_CLK2_SEL(1U):
+            freq = imxrt_clock_getoscfreq();
+            break;
+        case CCM_CBCMR_PERIPH_CLK2_SEL(2U):
+            freq = imxrt_clock_getpllfreq(kCLOCK_PllSys);
+            break;
+        case CCM_CBCMR_PERIPH_CLK2_SEL(3U):
+        default:
+            freq = 0U;
+            break;
+        }
+
+        freq /= (((CCM->CBCDR & CCM_CBCDR_PERIPH_CLK2_PODF_MASK) >> CCM_CBCDR_PERIPH_CLK2_PODF_SHIFT) + 1U);
+    } else { /* Pre_Periph_clk ---> Periph_clk */
+        switch (CCM->CBCMR & CCM_CBCMR_PRE_PERIPH_CLK_SEL_MASK) {
+        /* PLL2 ---> Pre_Periph_clk ---> Periph_clk */
+        case CCM_CBCMR_PRE_PERIPH_CLK_SEL(0U):
+            freq = imxrt_clock_getpllfreq(kCLOCK_PllSys);
+            break;
+        /* PLL2 PFD2 ---> Pre_Periph_clk ---> Periph_clk */
+        case CCM_CBCMR_PRE_PERIPH_CLK_SEL(1U):
+            freq = imxrt_clock_getsyspfdfreq(kCLOCK_Pfd2);
+            break;
+        /* PLL2 PFD0 ---> Pre_Periph_clk ---> Periph_clk */
+        case CCM_CBCMR_PRE_PERIPH_CLK_SEL(2U):
+            freq = imxrt_clock_getsyspfdfreq(kCLOCK_Pfd0);
+            break;
+        /* PLL1 divided(/2) ---> Pre_Periph_clk ---> Periph_clk */
+        case CCM_CBCMR_PRE_PERIPH_CLK_SEL(3U):
+            freq = imxrt_clock_getpllfreq(kCLOCK_PllArm) /
+                    (((CCM->CACRR & CCM_CACRR_ARM_PODF_MASK) >> CCM_CACRR_ARM_PODF_SHIFT) + 1U);
+            break;
+        default:
+            freq = 0U;
+            break;
+        }
+    }
+
+    return freq;
+}
+
+/*!
+ * brief Initialize the external 24MHz clock.
+ *
+ * This function supports two modes:
+ * 1. Use external crystal oscillator.
+ * 2. Bypass the external crystal oscillator, using input source clock directly.
+ *
+ * After this function, please call ref CLOCK_SetXtal0Freq to inform clock driver
+ * the external clock frequency.
+ *
+ * param bypassXtalOsc Pass in true to bypass the external crystal oscillator.
+ * note This device does not support bypass external crystal oscillator, so
+ * the input parameter should always be false.
+ */
+void imxrt_clock_initexternalclk(bool bypassXtalOsc)
+{
+    /* This device does not support bypass XTAL OSC. */
+    assert(!bypassXtalOsc);
+
+    CCM_ANALOG->MISC0_CLR = CCM_ANALOG_MISC0_XTAL_24M_PWD_MASK; /* Power up */
+    while ((XTALOSC24M->LOWPWR_CTRL & XTALOSC24M_LOWPWR_CTRL_XTALOSC_PWRUP_STAT_MASK) == 0) {
+    }
+    CCM_ANALOG->MISC0_SET = CCM_ANALOG_MISC0_OSC_XTALOK_EN_MASK; /* detect freq */
+    while ((CCM_ANALOG->MISC0 & CCM_ANALOG_MISC0_OSC_XTALOK_MASK) == 0) {
+    }
+    CCM_ANALOG->MISC0_CLR = CCM_ANALOG_MISC0_OSC_XTALOK_EN_MASK;
+}
+
+/*!
+ * brief Deinitialize the external 24MHz clock.

```

Comment:
same as above.
Please follow for all new API's added

Reply:
I fixed it
Please check it